### PR TITLE
Edit and destroy book

### PR DESCRIPTION
**Pull request by NgocVu1994**

### CODE REVIEW

File: app/controllers/admin/books_controller.rb [hunk 1]
```diff
@@ -1,4 +1,6 @@
 class Admin::BooksController < ApplicationController
+  before_action :load_book, except: [:index, :new, :create]
+  before_action :load_category, only: [:new, :update, :edit]
```
Comment (hellovietnam93) [inline app/controllers/admin/books_controller.rb]: có cần trong function update

File: app/controllers/admin/books_controller.rb [hunk 2]
```diff
@@ -1,4 +1,6 @@
 class Admin::BooksController < ApplicationController
+  before_action :load_book, except: [:index, :new, :create]
```
Comment (hellovietnam93) [inline app/controllers/admin/books_controller.rb]: read about cancancan

File: app/views/admin/books/index.html.erb [hunk 3]
```diff
@@ -24,9 +24,9 @@
         <td><%= book.content.truncate(50) %></td>
         <td><%= book.author %></td>
         <td>
-          <%= link_to t("admin.books.table.edit"), "#",
+          <%= link_to t("admin.books.table.edit"), edit_admin_book_path(book.id),
```
Comment (hellovietnam93) [inline app/views/admin/books/index.html.erb]: don't need id

File: app/views/admin/books/index.html.erb [hunk 4]
```diff
@@ -24,9 +24,9 @@
...
             class: "btn btn-primary" %>
-          <%= link_to t("admin.books.table.delete"), "#",
+          <%= link_to t("admin.books.table.delete"), admin_book_path(book.id),
```
Comment (hellovietnam93) [inline app/views/admin/books/index.html.erb]: the same as below

File: app/controllers/admin/books_controller.rb [hunk 5]
```diff
@@ -18,7 +18,36 @@ def create
     end
   end
 
+  def edit
+  end
+
+  def update
+    if @book.update_atrtributes book_params
+      flash[:success] = t("admin.books.updated_success")
```
Comment (ggbond2605) [inline app/controllers/admin/books_controller.rb]: remove ()

File: app/controllers/admin/books_controller.rb [hunk 6]
```diff
@@ -18,7 +18,36 @@ def create
...
+      redirect_to [:admin, @book]
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    if @book.destroy
+      flash[:success] = t("admin.books.destroy-success")
```
Comment (ggbond2605) [inline app/controllers/admin/books_controller.rb]: remove()

File: app/controllers/admin/books_controller.rb [hunk 7]
```diff
@@ -18,7 +18,36 @@ def create
...
+    else
+      flash[:alert] = t("admin.books.destroy-fail")
```
Comment (ggbond2605) [inline app/controllers/admin/books_controller.rb]: tương tự

File: app/controllers/admin/books_controller.rb [hunk 8]
```diff
@@ -18,7 +18,32 @@ def create
...
+    if @book.destroy
+      flash[:success] = t "admin.books.destroy-success"
+    else
+      flash[:alert] = t "admin.books.destroy-fail"
+    end
+    redirect_to [:admin, @book]
+  end
+
   private
+  def load_category
```
Comment (hellovietnam93) [inline app/controllers/admin/books_controller.rb]: load_categoríe

File: config/settings.yml [hunk 9]
```diff
@@ -5,6 +5,7 @@ books:
   page: 15
   size: "150x150"
   url: "/assets/anh1.jpg"
+  content: 50
```
Comment (hellovietnam93) [inline config/settings.yml]: it is used for what ?

File: app/views/admin/books/index.html.erb [hunk 10]
```diff
@@ -21,12 +21,12 @@
           <% end %>
         </td>
         <td><%= link_to book.title, "#" %></td>
-        <td><%= book.content.truncate(50) %></td>
+        <td><%= book.content.truncate(Settings.books.content) %></td>
```
Comment (hellovietnam93) [inline app/views/admin/books/index.html.erb]: remove ()

**Comment by hellovietnam93** [comment]
OK

**Pull request merged**